test/e2e/apps/rc.go

```diff
@@ -156,8 +162,15 @@ var _ = SIGDescribe("ReplicationController", func() {
 		framework.ExpectNoError(err, "failed to marshal json of replicationcontroller label patch")
 		// Patch the ReplicationController
 		ginkgo.By("patching ReplicationController")
-		_, err = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Patch(context.TODO(), testRcName, types.StrategicMergePatchType, []byte(rcLabelPatchPayload), metav1.PatchOptions{})
+		testRcPatched, err := f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Patch(context.TODO(), testRcName, types.StrategicMergePatchType, []byte(rcLabelPatchPayload), metav1.PatchOptions{})
 		framework.ExpectNoError(err, "Failed to patch ReplicationController")
+		framework.ExpectEqual(testRcPatched.ObjectMeta.Labels["test-rc"], "patched", "failed to patch RC")
+		ginkgo.By("waiting for RC to be modified")
+		for watchEvent := range rcWatchChan {
+			if watchEvent.Type == watch.Modified {
```

Comment:
what is the intent of this loop? other things (like the controller) could produce modified events on an RC besides the patch